Lib/tty.py

```diff
@@ -15,22 +24,81 @@
 OSPEED = 5
 CC = 6
 
-def setraw(fd, when=TCSAFLUSH):
-    """Put terminal into a raw mode."""
-    mode = tcgetattr(fd)
-    mode[IFLAG] = mode[IFLAG] & ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
-    mode[OFLAG] = mode[OFLAG] & ~(OPOST)
-    mode[CFLAG] = mode[CFLAG] & ~(CSIZE | PARENB)
-    mode[CFLAG] = mode[CFLAG] | CS8
-    mode[LFLAG] = mode[LFLAG] & ~(ECHO | ICANON | IEXTEN | ISIG)
+try:
+    CHECK_WINSZ = TIOCGWINSZ
+    CHECK_WINSZ = TIOCSWINSZ
+    HAVE_WINSZ = True
+
+    def getwinsz(fd):
+        """Gets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        s = pack("HHHH", 0, 0, 0, 0)
+        return ioctl(fd, TIOCGWINSZ, s)
+
+    def setwinsz(fd, winsz):
+        """Sets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        ioctl(fd, TIOCSWINSZ, winsz)
+except NameError:
+    HAVE_WINSZ = False
+    getwinsz = None
+    setwinsz = None
+
+def mode_echo(mode, echo=True):
```

Comment:
`mode_echo` is not the best name I guess.
Looking on the function body I can suggest renaming with `enable_echo()`.

`enable_echo(mode)` and `enable_echo(mode, False)`.
Maybe others have better idea?

Reply:
`enable_echo()` does sound better than `mode_echo()`. If you agree with `makeraw()` instead of `mode_raw()`, then we could also change this to `make_echo()`; however, I think that `enable_echo()` sounds better than either. I am leaning toward `enable_raw()`, and the updated commit will use that name unless someone else suggests something better.

Reply:
How about `mkecho()`? See next section.

Reply:
Sounds good